models/demos/llama3/lt

```diff
@@ -46,12 +46,39 @@ def ensure_ttsmi_installed():
     config_file = os.path.join(config_dir, f"reset_config_{hostname}.json")
 
     try:
-        subprocess.run(["tt-smi", "-g", config_file], check=True)
+        if "glx" in hostname:
+            generate_tg_config_file()
+        else:
+            subprocess.run(["tt-smi", "-g", config_file], check=True)
         print(f"Generated reset config file: {config_file}")
     except subprocess.CalledProcessError as e:
         print(f"Error generating reset config file: {e}")
         sys.exit(1)
 
+    print("Resetting device on program start...")
+    reset_device_sync(config_file)
+
+
+def reset_device_sync(config_file):
+    reset_cmd = ["tt-smi", "-r", config_file]
+    try:
+        print(f"Resetting device using config file: {config_file}")
+        result = subprocess.run(reset_cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+        print(f"Device reset successfully: {result.stdout}")
+    except subprocess.CalledProcessError as e:
+        print(f"Error during device reset: {e.stderr}")
+        sys.exit(1)
+
+
+def get_host_device():
+    hostname = os.environ.get("HOSTNAME", "unknown")
+    host_device = ""
+    if "t30" in hostname:
+        host_device = "T3K"
+    elif "glx" in hostname:
+        host_device = "TG"
+    return host_device
```

Comment:
the hostname is used to get the actual device. Shouldn't we just probe tt-smi instead?

For example `tt-smi -ls` will get you a list of devices:
```
                All available boards on host:                
┏━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃ PCI Dev ID ┃ Board Type ┃ Device Series ┃ Board Number    ┃
┡━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ 0          │ Wormhole   │ n300 L        │ 10001451172317c │
│ 1          │ Wormhole   │ n300 L        │ 100014511723132 │
│ 2          │ Wormhole   │ n300 L        │ 100014511723137 │
│ 3          │ Wormhole   │ n300 L        │ 10001421170300e │
│ N/A        │ Wormhole   │ n300 R        │ 10001451172317c │
│ N/A        │ Wormhole   │ n300 R        │ 100014511723132 │
│ N/A        │ Wormhole   │ n300 R        │ 100014511723137 │
│ N/A        │ Wormhole   │ n300 R        │ 10001421170300e │
└────────────┴────────────┴───────────────┴─────────────────┘
                  Boards that can be reset:                  
┏━━━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃ PCI Dev ID ┃ Board Type ┃ Device Series ┃ Board Number    ┃
┡━━━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ 0          │ Wormhole   │ n300 L        │ 10001451172317c │
│ 1          │ Wormhole   │ n300 L        │ 100014511723132 │
│ 2          │ Wormhole   │ n300 L        │ 100014511723137 │
│ 3          │ Wormhole   │ n300 L        │ 10001421170300e │
└────────────┴────────────┴───────────────┴─────────────────┘
```

Reply:
And since we use this function to get the actual device name, this doesn't work if you run `lt` on N150/N300 (I know it's uncommon).

Reply:
@johanna-rock-tt

Reply:
Yeah sounds good. I wasn't aware you can get the actual hostname from tt-smi and thought we only run it on t3k and tg, but with this we could just get the hostname and then show whatever fake devices are supported per actual device.